src/NCPoly.jl

```diff
@@ -102,6 +102,8 @@ number_of_variables(a::NCPolyRing) = 1
 
 characteristic(a::NCPolyRing) = characteristic(base_ring(a))
 
+is_finite(a::NCPolyRing) = false
```

Comment:
this is wrong in the case of the base ring `residue_ring(ZZ, 1)[1]` (aka the zero ring)

Reply:
Indeed. Note that then `characteristic` is also wrong, and I bet many other things.

To deal with this, I think we should add a method for "detecting" zero rings. In GAP the name would be a function named `is_trivial` (the definition for "trivial" is basically "has size 1" which works great for groups, vector spaces and rings ("trivial ring" is a synonym for "zero ring").

Then I could use that here -- but I wonder if it wouldn't be better to forbid matrix/polynomial/series/etc. rings over zero rings, simply because so many things are broken. (I.e. do the `is_trivial` check in the constructors). Hrm.

Reply:
Actually at least in some cases `characteristic` should be right (namely 1), and perhaps I could check just for that (a ring is trivial iff its characteristic is 1). At least this way we only have to get `characteristic` right.